src/components/fields/nested-multi-select/nested-multi-select.tsx

```diff
@@ -0,0 +1,154 @@
+import { L1OptionProps, L2OptionProps, L3OptionProps } from "@lifesg/react-design-system";
+import { Form } from "@lifesg/react-design-system/form";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import useDeepCompareEffect from "use-deep-compare-effect";
+import * as Yup from "yup";
+import { IGenericFieldProps } from "..";
+import { TestHelper } from "../../../utils";
+import { useValidationConfig } from "../../../utils/hooks";
+import { ERROR_MESSAGES } from "../../shared";
+import { IBaseOption, IL1Option, IL2Option, INestedMultiSelectSchema } from "./types";
+
+export const NestedMultiSelect = (props: IGenericFieldProps<INestedMultiSelectSchema>) => {
+	// =============================================================================
+	// CONST, STATE, REFS
+	// =============================================================================
+	const {
+		schema: { label, validation, options, ...otherSchema },
+		id,
+		value,
+		onChange,
+		error,
+		...otherProps
```

Comment:
we recently made some enhancements to the label, now all fields are injected with the `formattedLabel` prop, may I trouble you to add it as well, you can refer to: https://github.com/LifeSG/web-frontend-engine/blob/main/src/components/fields/text-field/text-field.tsx#L17